Tribler/community/chant/MDPackXDR.py

```diff
@@ -0,0 +1,68 @@
+import xdrlib
+from Tribler.community.chant.timeutils import time2float, float2time
+from Tribler.pyipv8.ipv8.keyvault.crypto import ECCrypto
+
+INFOHASH_SIZE = 20 # bytes
+REGULAR_TORRENT = 1
+CHANNEL_TORRENT = 2 # version
+MD_DELETE = 3
+
+crypto = ECCrypto()
+
+def serialize_metadata_gossip(md, key=None):
+    p = xdrlib.Packer()
+
+    p.pack_int(                   md["type"])
+    p.pack_opaque(                md["public_key"])
+    p.pack_double(time2float(     md["timestamp"]))
+    p.pack_uhyper(                md["tc_pointer"]) # TrustChain pointer
+    if md["type"] == MD_DELETE:
+        p.pack_opaque(                md["delete_sig"])
+    else:
+        p.pack_fopaque(INFOHASH_SIZE, md["infohash"])
+        p.pack_uhyper(                md["size"])
+        p.pack_double(time2float(     md["torrent_date"]))
+        p.pack_string(                md["title"])
+        p.pack_string(                md["tags"])
+
+    if key:
+        #assert("sig" not in md)
+        # Now we sign it
+        sig = crypto.create_signature(key, p.get_buf())
+        p.pack_opaque(sig)
+        md["sig"] = sig
+    else:
+        assert("sig" in md)
```

Comment:
Please no asserts. Do a proper check if this is required and throw/handle the error properly.

Reply:
Ok, it was just a placeholder. At the moment of writing this, I have not yet decided if I really wanted to put the signature checking code inside the serialization procedure.
Now I can tell its probably gonna stick there, so I can safely add the error handling.